lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -1686,6 +1688,82 @@ class DecomposeAtenPadOp : public OpRewritePattern<AtenPadOp> {
 };
 } // namespace
 
+namespace {
+// Decompose `aten.split` op into `aten.slice` and `prim.ListConstruct` op.
+template <typename OpTy>
+class DecomposeAtenSplitLikeOp : public OpRewritePattern<OpTy> {
+public:
+  using OpRewritePattern<OpTy>::OpRewritePattern;
+  LogicalResult matchAndRewrite(OpTy op,
+                                PatternRewriter &rewriter) const override {
+    SmallVector<int64_t> offsets;
+    // TODO: Support non static offsets.
+    if (!matchPattern(op.split_sizes(), m_TorchConstantIntList(offsets)))
+      return rewriter.notifyMatchFailure(op, "constant split sizes supported");
+
+    int64_t dim;
+    if (!matchPattern(op.dim(), m_TorchConstantInt(&dim)))
+      return rewriter.notifyMatchFailure(op, "dim must be constant");
+
+    auto tensorType = op.self().getType().template dyn_cast<BaseTensorType>();
+    SmallVector<int64_t> sizes;
+    for (unsigned i = 0; i < tensorType.getSizes().size(); i++)
+      sizes.push_back(-1);
+
+    auto listType = op.getType().template cast<Torch::ListType>();
+    Location loc = op.getLoc();
+    SmallVector<int64_t> endIdx;
+    endIdx.push_back(offsets[0]);
+    for (unsigned i = 1; i < offsets.size(); i++) {
+      endIdx.push_back(offsets[i] + endIdx[i - 1]);
+    }
+    int64_t start = 0;
+    SmallVector<Value> listOfTensors;
+    Value stepVal = rewriter.create<Torch::ConstantIntOp>(
+        loc, rewriter.getI64IntegerAttr(1));
+    for (unsigned i = 0; i < offsets.size(); i++) {
+      Value startVal = rewriter.create<Torch::ConstantIntOp>(
+          loc, rewriter.getI64IntegerAttr(start));
+      Value endVal = rewriter.create<Torch::ConstantIntOp>(
+          loc, rewriter.getI64IntegerAttr(endIdx[i]));
+      sizes[dim] = endIdx[i] - start;
+      Type resultType = tensorType.getWithSizesAndDtype(
+          llvm::makeArrayRef(sizes), tensorType.getDtype());
+      Value slice = rewriter.create<AtenSliceTensorOp>(
+          loc, resultType, op.self(), op.dim(), startVal, endVal, stepVal);
+      Value sliceCast = rewriter.create<TensorStaticInfoCastOp>(
+          loc, listType.getContainedType(), slice);
+      sliceCast.dump();
+      listOfTensors.push_back(sliceCast);
+      start = endIdx[i];
+    }
+    // Value result = rewriter.create<PrimListConstructOp>(loc,
+    // Torch::ListType::get(resultType), listOfTensors);
+    // rewriter.replaceOpWithNewOp<TensorStaticInfoCastOp>(op, op.getType(),
+    //                                                     result);
+    Value result = rewriter.create<PrimListConstructOp>(
+        loc, Torch::ListType::get(listType.getContainedType()), listOfTensors);
+    result.dump();
+    rewriter.replaceOp(op, result);
+    llvm::errs() << "Done till here\n";
+    return success();
+  }
+};
+} // namespace
+
+namespace {
+class DecomposeAtenSplitSizesOp : public OpRewritePattern<AtenSplitSizesOp> {
```

Comment:
Does `patterns.add<DecomposeAtenSplitLikeOp<AtenSplitSizesOp>>(context);` not work?